addons/html_builder/static/src/builder/plugins/builder_options_plugin.js

```diff
@@ -55,14 +55,31 @@ export class BuilderOptionsPlugin extends Plugin {
         }
 
         const previousElementToIdMap = new Map(this.lastContainers.map((c) => [c.element, c.id]));
-        this.lastContainers = [...elementToOptions]
+        const newContainers = [...elementToOptions]
             .sort(([a], [b]) => (b.contains(a) ? 1 : -1))
             .map(([element, options]) => ({
                 id: previousElementToIdMap.get(element) || uniqueId(),
                 element,
                 options,
                 isRemovable: isRemovable(element),
             }));
+
+        // Do not update the containers if they did not change.
+        if (newContainers.length === this.lastContainers.length) {
+            const previousIds = this.lastContainers.map((c) => c.id);
+            if (newContainers.every((c) => previousIds.includes(c.id))) {
+                const previousOptions = this.lastContainers.map((c) => c.options).flat();
+                const newOptions = newContainers.map((c) => c.options).flat();
+                if (newOptions.length === previousOptions.length) {
+                    const previousOptionIds = previousOptions.map((option) => option.id);
+                    if (newOptions.every((option) => previousOptionIds.includes(option.id))) {
+                        return;
+                    }
+                }
+            }
+        }
```

Comment:
What about the following code for lisibility?
```js
        const optionsIds = newContainers.map((c) => c.options.map((o) => o.id)).flat();
        let isDifferent = this.lastOptionIds.length !== optionsIds.length;
        if (!isDifferent) {
            for (let i = 0; i < this.lastOptionIds.length; i++) {
                if (this.lastOptionIds[i] !== optionsIds[i]) {
                    isDifferent = true;
                    break;
                }
            }
        }
```

`this.lastOptionIds` needs to be initialized to `[]`

Reply:
I still need to check if the elements ids didn't change too, because the options id do not always change, for example, if I click on a column and then on another similar one (meaning they have the same options), the options have the same ids, so with your suggestion it won't update the container, but it should since I clicked  on another element :slightly_smiling_face: 

I will still try to make the checks look better, given that we won't have that many options/elements, no need to try to optimize it that much :slightly_smiling_face: